src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -72,6 +72,43 @@ function showTooltip($input, theme, message) {
     $input.data('$tooltip').show();
 }
 
+/**
+ * Validate the input for decimal values.
+ *
+ * This function ensures that the input value is either empty or follows
+ * the rules for decimal numbers. It allows numbers with optional
+ * thousands separators (commas) and a mandatory decimal point (dot).
+ *
+ * @param {jQuery} $input
+ */
+function validateDecimalInput ($input) {
+    const value = converter.convert($input.val());
+    const thousandsSeparator = locale.getThousandsSeparator();
+    const decimalSeparator = locale.getDecimalSeparator();
+
+    const escapedThousandsSeparator = thousandsSeparator.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    const escapedDecimalSeparator = decimalSeparator.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+    const regexPattern = thousandsSeparator
+        ? `^$|^-?\\d{1,3}(${escapedThousandsSeparator}\\d{3})*(${escapedDecimalSeparator}\\d+)?$|^-?\\d+(${escapedDecimalSeparator}\\d+)?$|^-?\\d*${escapedDecimalSeparator}$|^-?${escapedDecimalSeparator}\\d+$`
+        : `^$|^-?\\d+(${escapedDecimalSeparator}\\d+)?$|^-?\\d*${escapedDecimalSeparator}$|^-?${escapedDecimalSeparator}\\d+$`;
+
+    const regex = new RegExp(regexPattern);
+
+    if (!regex.test(value)) {
+        $input.addClass('invalid');
+        $input.addClass('error');
+        const decimalError = thousandsSeparator
+            ? __('Invalid value, use %s (dot) for decimal point and %s (comma) for thousands separator.', decimalSeparator, thousandsSeparator)
```

Comment:
Still not quite right. If separators were set from `ru-RU`, text would be: "_Invalid value, use , (dot) for decimal point and . (comma) for thousands separator._". Symbol-name should correspond to the symbol. In case of unknown symbol, write no symbol-name at all.